src/inausoft.netCLI/CLIConfiguration.cs

```diff
@@ -0,0 +1,41 @@
+﻿using inausoft.netCLI.Deserialization;
+using System;
+using System.Collections.Generic;
+
+namespace inausoft.netCLI
+{
+    public class CLIConfiguration
+    {
+        internal readonly Dictionary<Type, Type> _commandMap;
```

Comment:
I felt little bit uneasy usage of this internal field. 
What i meant is that I the usage `cliConfiguration._commandMap` seems little bit odd. 
On the other hand, switching it into property would cause removing `readonly` keyword, which seems importand in this situation. I don't know, maybe let it be ;)

Reply:
@xHoody - any ideas?